### ISSUE TITLE
Regression: Caja is hyphenating long filenames again

### ISSUE THREAD

**Issue by qazwsxedcgit** [mate-desktop/caja#1583]
#### Expected behaviour
Caja should not insert spurious hyphens into filenames.

#### Actual behaviour
Caja does insert spurious hyphens into filenames, which is extremely confusing. This is a regression from issue #1284 .

#### Steps to reproduce the behaviour
Create a file with a name long enough so that it wraps in icon view, for example "thisisaverylongfilename.txt"
Switch to icon view in Caja. The filename will be displayed below the file's icon like this:
thisisaverylongfi-
lename.txt

Sample image attached.

This bug severely affects usability because it makes it impossible to see whether a file is named e.g. "filename" or "file-name". 

With reference to the fix mentioned in #1320, which refers to pango 1.44:
libpango in the affected Debian 11 system is at version 1.46.2-3
It therefore appears that this is a regression to the unfortunate design choice in pango to hyphenate file names. If there is a way to override this behaviour in Caja's code instead of relying on pango, I urge that it be considered.

#### MATE general version
1.24.1

#### Package version
Caja 1.24.0
libpango in the affected Debian 11 system is at version 1.46.2-3

#### Linux Distribution
Debian 11 (Bullseye)

![SpuriousHyphen](https://user-images.githubusercontent.com/6451926/141538421-8d65cd35-7429-4d9c-b322-0a4c44074d93.png)

**Comment by lucas-oliver-trondsen** [comment]
I totally agree with qazwsxedcgit. Please stop Caja from inserting hyphens into file names.

**Comment by lukefromdc** [comment]
Pango seems to be getting worse and worse about this. Last time around, we had to hardcode the cboice of an option not to auto-hypenate into Caja. This mess also appears anywhere else libpango is used. The only reason I do not force-install and then lock an older version of pango (or compile a modded version wirh this removed) is because I have to be able to test this stuff. I don't EVER want auto-anything with text processing.

Not been around my desktop as much though since the chaos started in my country.

**Comment by lukefromdc** [comment]
I've got libpango 1.48 and on a very long filename did not get added hyphens in a test tonight either in a navigation window or on the desktop

**Comment by guruthree** [comment]
For what it's worth, I can provide another data point. My Gentoo install recently updated pango from 1.42.4 to 1.48.10 and hyphenation appeared in filenames. Downgrading back to 1.42.4 removed the hyphenation.

**Comment by lukefromdc** [comment]
Caja needs to be rebuilt when the newer versions of pango are installed. The code to stop unwanted hyphenation as I recall calls a function only available in those versions of pango and thus is only built when one of those versions of pango is installed on the build system. Running caja built against this with an older pango version I would expect to crash on the resulting undefined symbol.

 Away from my desktop now though, so working from memory and cannot double-check this.

**Comment by guruthree** [comment]
It looks like rebuilding caja after updating pango removed the hyphenations for me on Gentoo.

**Comment by qazwsxedcgit** [comment]
For those who rely on binaries packaged by a distribution, rebuilding from source with a different pango isn't an option. If only certain combinations of <version of caja> and <version of libpango> avoid adding hyphens to filenames, then this needs documenting and passing on to distribution maintainers. Alternatively, is there a way to hardcode desired hyphenation behaviour into caja so that the exact version of libpango no longer matters for this?

**Comment by lukefromdc** [comment]
I've never heard of a runtime check of a package version as a means of avoiding undefined symbol errors. This is something distros will need to keep up on, though rolling releases would be expected to be the main problem E.g
 .a new Ubuntu version will mean a new Caja versions and thus a new build. Not so on Debian Sid.

**Comment by lukefromdc** [comment]
Also remember that Gentoo is a special case, being a source-based distro that need not do anything on the distro end when packages need to be rebuilt but not rewritten. Distros need to be on top of library changes (which the libraries should document) that force rebuilding packages.  

If only Caja or only Nemo or only Nautilus documented this one for instance, the other two would not be rebuilt by distros. If the Pango devs on releasing the first version with autohyphenation and the option to turn it off got the documentation right, then it would be on distros to check every package depending on pango or simply issue rebuilds of all of them, whichever is easier for them.

I for one play no role in communicating with distros, those on the team that do should give the distros they are involved with a heads-up at this point, as this transition is not a new one and we are seeing problems again.

**Comment by lukefromdc** [comment]
As for hardcoding directly into Caja, that would probably require reimplementing some (or even all) of Pango's work ourselves and maintaining it. Alternately in build defaults static linking a  copy of a known good version of pango, then of course we have to monitor libpango for any critical bugs that force security or stability updates and remake the package at that point. That kind of solution leads to bloat and maintainance issues and can get ugly fast. OK for a package like a snap or flatpack but for upstream to force this model would mean taking responsibility for everything in the package

**Comment by bdantas** [comment]
I'm on Devuan 4 (Debian 11 sans systemd) and this issue was driving me crazy. Using a different version of caja or pango is not an option on a debianoid system with MATE because of a tight web of interdependencies.

What I did instead was to compile caja 1.24.0 on a lean Tiny Core Linux machine that doesn't suffer from this pango-induced hyphenation craziness. It turns out the caja binary works just fine on Devuan 4 (and presumably also on Debian 11, although I haven't tested it) except for three missing libraries.

If you'd like to stick with Devuan 4 / Debian 11, MATE 1.24.1, and  pango 1.46, feel free to try my solution. All you need is to do this:
```
sudo mv /usr/bin/caja /usr/bin/caja-insane
```
Then put caja.bin, caja.lib (directory containing the missing libraries), and caja (a wrapper script with simple LD_LIBRARY_PATH hack) in /usr/bin. Reboot and you'll be cured of the hyphenation issue :)

caja.bin, caja.lib, and the caja wrapper script are available on my personal server here: http://gnuser.ddns.net/public/caja

Cheers!

**Comment by bdantas** [comment]
P.S. If you have caja extensions installed (e.g., caja-open-terminal) then you need to do this in order for the extensions to work with my hack:
```
sudo mkdir -p /usr/lib/caja
sudo ln -s /usr/lib/x86_64-linux-gnu/caja/extensions-2.0/ /usr/lib/caja/extensions-2.0
```
Followed by a reboot or `pkill -f caja`
